### PR TITLE
[embedded-wallet] Conform to EIP 1193

### LINKED ISSUE #248: [embedded-wallet] Implement EIP1193-compliant public API
Currently, the Embedded Wallet lives as a global called `EmbeddedWallet`, with `enable` and `request` methods. This isn't entirely compliant with [EIP 1193](https://github.com/ryanio/EIPs/blob/master/EIPS/eip-1193.md), the standard for APIs that operate as Ethereum Providers in JavaScript.

This would imply:

- [x] Rename the global to `channelProvider` (6004ae2)
- [x] Refactor `request` into `send(method, params)` (7cbe92b)
- [x] Implement `subscribe` and `unsubscribe` via EventEmitter (5d7bece)
- [x] ~Implement `connect`, `close`, `networkChanged`, `accountsChanged` events (leave these last two as placeholders)~
- [x] Make `enable` async (ac8ad78)

Work is being done in #250.

### PR DESCRIPTION
**Pull request by joelalejandro**
Resolves #248.

Also refactors Web3Torrent's usage of Embedded Wallet to conform to the updated API.

See the updated [wallet.js](https://github.com/statechannels/monorepo/blob/embedded-wallet/eip-1193/packages/embedded-wallet/public/wallet.js#L143) and [README](https://github.com/statechannels/monorepo/blob/embedded-wallet/eip-1193/packages/embedded-wallet/README.md) to understand how the API has changed.

We're now including an inline version of EventEmitter so we don't have to load another external dependency. This decision can be revisited later.

### CODE REVIEW

File: packages/embedded-wallet/public/wallet.js [hunk 1]
```diff
@@ -1,164 +1,210 @@
...
+  return new Promise((resolve, reject) => {
+    let listener;
+    if (callback) {
+      listener = event => {
+        if (event.data && event.data.jsonrpc && event.data.result && event.data.id === message.id) {
+          callback(event.data.result);
+          window.removeEventListener('message', listener);
+          log('Received response: %o', event.data);
+          resolve(event.data.result);
+        } else if (event.data.error) {
```
Comment (spersico) [inline packages/embedded-wallet/public/wallet.js]: Unless I'm reading this wrong, you are not checking if event.data is defined

File: packages/embedded-wallet/public/wallet.js [hunk 2]
```diff
@@ -1,164 +1,210 @@
...
+          reject(event.data.error);
+        }
+      };
+    } else {
+      listener = event => {
+        if (event.data && event.data.jsonrpc && event.data.result && event.data.id === message.id) {
+          window.removeEventListener('message', listener);
+          log('Received response: %o', event.data);
+          resolve(event.data.result);
+        } else if (event.data.error) {
```
Comment (spersico) [inline packages/embedded-wallet/public/wallet.js]: Same here. I would add a `|| !event.data`

**Comment by joelalejandro** [comment]
I'll keep this open so @ebryn has a chance to review.

**Comment by joelalejandro** [comment]
As soon as checks pass again, I'm merging this so we can move forward. We can always revisit any decisions made in this PR.

**Pull request merged**